python-recipes/vector-search/02_hybrid_search.ipynb

```diff
@@ -0,0 +1,1205 @@
+    "\n",
+    "# Add the optional flag, \"~\", so that this doesn't also act as a strict text filter\n",
+    "full_text_query = f\"(~{base_full_text_query})\"\n",
+    "\n",
+    "\n",
+    "# Add full-text predicate to the vector query \n",
+    "query.set_filter(full_text_query)\n",
+    "query.query_string()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**The query string above combines both full-text search and a vector search.** This will be passed to the aggregation API to combine using a simple weighted sum of scores before a final sort and truncation.\n",
+    "\n",
+    "Below, we extend the `redis-py` aggregation request class to support setting the full-text scoring algorithm. This should be available out-of-the-box very soon."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 74,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Any, Dict, List\n",
+    "from redis.commands.search.aggregation import AggregateRequest, Desc\n",
+    "\n",
+    "# TODO: .scorer attribute not yet available in redis-py yet for aggregations\n",
+    "# This is in a PR for the library and should be available soon\n",
+    "# For now, we will extend the class and implement the method directly for demonstration\n",
+    "\n",
+    "class AggregateRequestWithScorer(AggregateRequest):\n",
+    "    _scorer: str = None\n",
+    "\n",
+    "    def scorer(self, scorer: str):\n",
+    "        self._scorer = scorer\n",
+    "        return self\n",
+    "\n",
+    "    def build_args(self):\n",
+    "        # @foo:bar ...\n",
+    "        ret = [self._query]\n",
+    "\n",
+    "        if self._scorer:\n",
+    "            ret.extend([\"SCORER\", self._scorer])\n",
+    "\n",
+    "        if self._with_schema:\n",
+    "            ret.append(\"WITHSCHEMA\")\n",
+    "\n",
+    "        if self._verbatim:\n",
+    "            ret.append(\"VERBATIM\")\n",
+    "\n",
+    "        if self._add_scores:\n",
+    "            ret.append(\"ADDSCORES\")\n",
+    "\n",
+    "        if self._cursor:\n",
+    "            ret += self._cursor\n",
+    "\n",
+    "        if self._loadall:\n",
+    "            ret.append(\"LOAD\")\n",
+    "            ret.append(\"*\")\n",
+    "        elif self._loadfields:\n",
+    "            ret.append(\"LOAD\")\n",
+    "            ret.append(str(len(self._loadfields)))\n",
+    "            ret.extend(self._loadfields)\n",
+    "\n",
+    "        if self._dialect:\n",
+    "            ret.extend([\"DIALECT\", self._dialect])\n",
+    "\n",
+    "        ret.extend(self._aggregateplan)\n",
+    "\n",
+    "        return ret"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Run the query!\n",
+    "\n",
+    "Notes on aggregate query syntax \n",
+    "- `.scorer`: specifies the scoring function to use either BM25 or TFIDF\n",
```

Comment:
I think we support a few more than BM25 or TFIDF. Up to you if you want to list them all.

Reply:
Good note! updated to link to docs where it lists them all